torchdynamo/variables/functions.py

```diff
@@ -263,6 +282,23 @@ def call_function(
         return result
 
 
+def invoke_and_store_as_constant(tx, fn, name, source, options, args, kwargs):
+    def convert(x):
+        if isinstance(x, variables.TensorVariable):
+            return x.proxy.node.meta["example_value"]
+        return x.as_python_constant()
+
+    args = [convert(x) for x in args]
+    kwargs = {k: convert(x) for k, v in kwargs.items()}
+    res = fn(*args, **kwargs)
+    return tx.output.add_submodule(
+        res,
+        name,
+        source=NNModuleSource(GetItemSource(source, name)),
```

Comment:
This source isn't right.

`source` is how we can retrieve an input value from user data structures.

Reply:
Good catch, ty.